tests/decorators/test_prompt_decorator.py

```diff
@@ -0,0 +1,259 @@
+        # provider calls. Could not find a way to intercept them coming from a Mock.
+        client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))  # type: ignore
+        return (
+            client.chat.completions.create(
+                model=model,
+                messages=messages,  # type: ignore
+                temperature=0.8,
+            )
+            .choices[0]
+            .message.content
+        )
+    if provider == "anthropic":
+        client = Anthropic(api_key=os.getenv("ANTHROPIC_API_KEY"))  # type: ignore
+        messages_anthropic_format = [
+            MessageParam(
+                content=message["content"],
+                role="user" if message["role"] in ("user", "system") else "assistant",
+            )
+            for message in messages
+        ]
+        return (
+            client.messages.create(  # type: ignore
+                model=model,
+                messages=messages_anthropic_format,
+                max_tokens=200,
+                temperature=0.8,
+            )
+            .content[0]
+            .text
+        )
+    if provider == "groq":
+        try:
+            client = Groq(  # type: ignore
+                # This is the default and can be omitted
+                api_key=os.environ.get("GROQ_API_KEY"),
+            )
+            return (
+                client.chat.completions.create(
+                    messages=messages,  # type: ignore
+                    model=model,
+                    temperature=0.8,
+                )
+                .choices[0]
+                .message.content
+            )
+        except GroqNotFoundError:
+            # NOTE: Tests in this file are integration tests that rely on live LLM provider
+            # clients. If a test fails, it might be flaky. If this happens, consider adding
+            # a skip mechanism similar to Groq
+            pytest.skip("GROQ not available")
+    if provider == "cohere":
+        client = cohere.Client(api_key=os.getenv("COHERE_API_KEY"))  # type: ignore
+        messages_cohere_format: list[cohere.Message] = []
+        for message in messages:
+            if message["role"] == "system":
+                messages_cohere_format.append(cohere.SystemMessage(message=message["content"]))
+            elif message["role"] == "user":
+                messages_cohere_format.append(cohere.UserMessage(message=message["content"]))
+            elif message["role"] == "assistant":
+                messages_cohere_format.append(cohere.ChatbotMessage(message=message["content"]))
+        return client.chat(  # type: ignore
+            chat_history=messages_cohere_format,
+            model=model,
+            max_tokens=200,
+            message=messages[-1]["content"],
+            temperature=0.8,
+        ).text
+    if provider == "replicate":
+        # TODO: Instrumentor only picks up methods on module-level, not client level
```

Comment:
Ik this is problematic but it's an Instrumentor problem. Best solution would be to do a PR against the open source library. Otherwise, we should document behaviour on our side